lib/metadataTypes/Asset.js

```diff
@@ -1181,6 +1180,69 @@ class Asset extends MetadataType {
         typeDirArr.push(templateName);
         return await File.readFile([templateDir, ...typeDirArr], fileName, 'json');
     }
+    /**
+     * should return only the json for all but asset, query and script that are saved as multiple files
+     * additionally, the documentation for dataExtension and automation should be returned
+     *
+     * @param {string[]} keyArr customerkey of the metadata
+     * @returns {string[]} list of all files that need to be committed in a flat array ['path/file1.ext', 'path/file2.ext']
+     */
+    static async getFilesToCommit(keyArr) {
+        const basePath = File.normalizePath([
+            this.properties.directories.retrieve,
+            this.buObject.credential,
+            this.buObject.businessUnit,
+        ]);
+
+        const fileList = (
+            await Promise.all(
+                keyArr.map(async (key) => {
+                    let subType;
+                    let filePath;
+                    let fileName;
+                    for (const st of this.definition.subTypes) {
+                        fileName = `${key}.${this.definition.type}-${st}-meta.json`;
+                        if (
+                            File.existsSync(
```

Comment:
prefer async

Reply:
Will fix

Reply:
Merged your change in, so found a handful of places that need updating afterwards.  Will do that here and mark it with your issue

Reply:
done